ouroboros-network/src/Ouroboros/Network/ChainFragment.hs

```diff
@@ -509,7 +510,8 @@ selectPointsSpec offsets c =
 -- | \( O(\log(\min(i,n-i)) \). Find the block after the given point.
 successorBlock :: HasHeader block
                => Point block -> ChainFragment block -> Maybe block
-successorBlock p c = case lookupBySlotFT c (pointSlot p) of
+successorBlock GenesisPoint           c = head c
```

Comment:
This is wrong, say you have a fragment (using `SlotNo`s for blocks) `[4, 5, 6]`, then the `successorBlock` of `GenesisPoint` should return `Nothing` instead of `head c = 6`. I'd change it to (note the extra sentence in the docstring):

```haskell
-- | \( O(\log(\min(i,n-i)) \). Find the block after the given point.
--
-- If the given point is not on the fragment (e.g., the genesis point), the
-- result will be 'Nothing'.
successorBlock :: HasHeader block
               => Point block -> ChainFragment block -> Maybe block
successorBlock GenesisPoint           _ = Nothing
successorBlock p@(BlockPoint bslot _) c = case lookupBySlotFT c bslot of
  FT.Position _ b ft'
    | blockPoint b == p
    , n FT.:< _ <- FT.viewl ft' -- O(1)
    -> Just n
  _ -> Nothing
```

Reply:
Ah yes this is definitely wrong. But shouldn't the successor of the genesis point be the oldest block in the fragment?

Reply:
Oh, but the genesis point is never on a fragment, so right let's do Nothing.